pkg/controllers/management/clusterprovisioner/provisioner.go

```diff
@@ -126,7 +126,10 @@ func (p *Provisioner) update(cluster *v3.Cluster, create bool) (*v3.Cluster, err
 		return cluster, err
 	}
 
-	v3.ClusterConditionProvisioned.True(cluster)
+	obj, err := v3.ClusterConditionProvisioned.Do(cluster, func() (runtime.Object, error) {
```

Comment:
probably clearer if you just set message and reason to "".

Reply:
@ibuildthecloud updated